tidy3d/web/api/webapi.py

```diff
@@ -436,32 +436,49 @@ def monitor_preprocess() -> None:
+    if verbose:
         # verbose case, update progressbar
         console.log("running solver")
-        with Progress(console=console) as progress:
-            pbar_pd = progress.add_task("% done", total=100)
-            perc_done, _ = get_run_info(task_id)
+        if task_type == "FDTD":
+            with Progress(console=console) as progress:
+                pbar_pd = progress.add_task("% done", total=100)
+                perc_done, _ = get_run_info(task_id)
+
+                while (
+                    perc_done is not None and perc_done < 100 and get_status(task_id) == "running"
+                ):
+                    perc_done, field_decay = get_run_info(task_id)
+                    new_description = f"solver progress (field decay = {field_decay:.2e})"
+                    progress.update(pbar_pd, completed=perc_done, description=new_description)
+                    time.sleep(RUN_REFRESH_TIME)
 
-            while perc_done is not None and perc_done < 100 and get_status(task_id) == "running":
                 perc_done, field_decay = get_run_info(task_id)
-                new_description = f"solver progress (field decay = {field_decay:.2e})"
-                progress.update(pbar_pd, completed=perc_done, description=new_description)
-                time.sleep(RUN_REFRESH_TIME)
-
-            perc_done, field_decay = get_run_info(task_id)
-            if perc_done is not None and perc_done < 100 and field_decay > 0:
-                console.log(f"early shutoff detected at {perc_done:1.0f}%, exiting.")
+                if perc_done is not None and perc_done < 100 and field_decay > 0:
+                    console.log(f"early shutoff detected at {perc_done:1.0f}%, exiting.")
 
-            new_description = f"solver progress (field decay = {field_decay:.2e})"
-            progress.update(pbar_pd, completed=100, refresh=True, description=new_description)
-
-    # TODO: progressbar for EME
+                new_description = f"solver progress (field decay = {field_decay:.2e})"
+                progress.update(pbar_pd, completed=100, refresh=True, description=new_description)
+        elif task_type == "EME":
+            with Progress(console=console) as progress:
+                pbar_pd = progress.add_task("% done", total=100)
+                perc_done, _ = get_run_info(task_id)
+
+                while (
+                    perc_done is not None and perc_done < 100 and get_status(task_id) == "running"
+                ):
+                    perc_done, _ = get_run_info(task_id)
+                    new_description = "solver progress"
+                    progress.update(pbar_pd, completed=perc_done, description=new_description)
+                    time.sleep(RUN_REFRESH_TIME)
+
+                perc_done, _ = get_run_info(task_id)
+                new_description = "solver progress"
+                progress.update(
+                    pbar_pd, completed=perc_done, refresh=True, description=new_description
```

Comment:
In the FDTD case, we set `completed=100` here to force the progress bar to look "done" once the solver is done. Any reason you set `perc_done` instead? Which I think should have no effect then?

Reply:
thanks for the catch!